### DISCUSSION TITLE
MMALCAM stuff and the Raspberry Pi

### DISCUSSION THREAD

**Discussion by git-user44** [Motion-Project/motion#1771] [Q&A]
I am running motion on a Raspberry Pi 2 and the camera is a NoIR Camera V2.

I have had it working for several years, the distro was buildroot-2016.08.1 & I think the version of motion running was 3.1.8

I recently did a software upgrade and my distro is now buildroot-2024.02.1 with motion being version 4.6.0.

OK. So lots of things have changed. Most notable is that my pictures have changed from a blueish tint to mauve. Now I know that this is because the IR filter is missing, but I remember in the dim distant past, that I fed some parameters into mmal that could alter the tint, and I ended up with something blueish. I used to have ...

mmalcam_name vc.ril.camera
mmalcam_control_params -ex night
mmalcam_control_params -ex auto -awb off -awbg 0.4,3

and I now have ...

mmalcam_params "mmalcam_name"=vc.ril.camera
mmalcam_params -ex night
mmalcam_params -ex auto -awb off -awbg 0.4,3

I'm not 100% sure of the format of the mmalcam_params command. I tried wrapping the parameters in quotes but it made no difference. I'm not sure what these parameters do as it was almost 9 years ago that I was tinkering with this stuff. What concerns me is that I can put any old rubbish in these commands and it makes no difference at all. Could someone confirm the syntax of the mmalcam_params command as I cannot find an example of it.

My other anomaly is that I read now that for the Raspberry Pi I have to start the program like "libcamerify motion". If I do that I get a segment fault as soon as movement is detected. If I start motion like before with the simple command "motion" then no segment faults happen and it behaves like before recording videos etc.

I see I can attach photos. I attach a before and after.

Thanks

![20240505080241-10](https://github.com/Motion-Project/motion/assets/22965693/da7d61da-7597-41c7-abba-ccbb6865804d)
![20240508112419-04](https://github.com/Motion-Project/motion/assets/22965693/10111d8d-ed04-4673-b609-8c68091cfb37)

**Comment by git-user44** [comment]
OK, so it has been over 3 weeks and no-one has answered, so I guess no-one knows.

I shall revert back to the software from 2016

**Comment by SpacedOutDude** [comment]
These are the settings I use
mmalcam_control_params -awb greyworld --ev 5 -sa 100

![3e31efe4](https://github.com/user-attachments/assets/606fcfe9-90ad-4d7a-9a11-9dbb0725c883)
